fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/CollectRelation.java

```diff
@@ -0,0 +1,190 @@
+                if (analyzedCte.isPresent()) {
+                    return;
+                }
+            }
+        }
+        List<String> tableQualifier = RelationUtil.getQualifierName(cascadesContext.getConnectContext(), nameParts);
+        TableIf table = cascadesContext.getConnectContext().getStatementContext()
+                .getAndCacheTable(tableQualifier, tableFrom);
+        if (collectMTMV) {
+            collectMTMVCandidates(table, cascadesContext);
+        }
+        if (table instanceof View) {
+            processView((View) table, cascadesContext);
+        }
+    }
+
+    private void collectMTMVCandidates(TableIf table, CascadesContext cascadesContext) {
+        if (cascadesContext.getConnectContext().getSessionVariable().enableMaterializedViewRewrite) {
+            Set<MTMV> mtmvSet = Env.getCurrentEnv().getMtmvService().getRelationManager()
+                    .getAvailableMTMVs(Lists.newArrayList(new BaseTableInfo(table)),
+                            cascadesContext.getConnectContext(),
+                            false, ((connectContext, mtmv)
+                                    -> MTMVUtil.mtmvContainsExternalTable(mtmv)
+                                    && !connectContext.getSessionVariable()
+                                    .isEnableMaterializedViewRewriteWhenBaseTableUnawareness()));
+            for (MTMV mtmv : mtmvSet) {
+                cascadesContext.getStatementContext().getMtmvRelatedTables().put(mtmv.getFullQualifiers(), mtmv);
+                mtmv.readMvLock();
+                try {
+                    for (BaseTableInfo baseTableInfo : mtmv.getRelation().getBaseTables()) {
+                        cascadesContext.getStatementContext().getAndCacheTable(baseTableInfo.toList(), TableFrom.MTMV);
+                    }
+                } finally {
+                    mtmv.readMvUnlock();
+                }
+            }
+        }
+    }
+
+    private void processView(View view, CascadesContext cascadesContext) {
+        String inlineViewDef = view.getInlineViewDef();
+        parseAndCollectFromView(inlineViewDef, cascadesContext);
+    }
+
+    private void parseAndCollectFromView(String ddlSql, CascadesContext parentContext) {
+        LogicalPlan parsedViewPlan = new NereidsParser().parseSingle(ddlSql);
+        // TODO: use a good to do this, such as eliminate UnboundResultSink
+        if (parsedViewPlan instanceof UnboundResultSink) {
+            parsedViewPlan = (LogicalPlan) ((UnboundResultSink<?>) parsedViewPlan).child();
+        }
+        CascadesContext viewContext = CascadesContext.initContext(
+                parentContext.getStatementContext(), parsedViewPlan, PhysicalProperties.ANY);
+        viewContext.keepOrShowPlanProcess(parentContext.showPlanProcess(),
+                () -> viewContext.newTableCollector().collect());
+        parentContext.addPlanProcesses(viewContext.getPlanProcesses());
+    }
```

Comment:
parsedViewPlan  not replace to the outer logical plan after parse and lead to parse view again?

Reply:
done

Reply:
for processing view in sub query expr will intro too many code and could intro bug easily here, so do not replace view with plan. instead, save view's sql here to avoid view's def changed